src/cmds.c

```diff
@@ -772,6 +775,61 @@ static void cmd_pls_bot(struct userrec *u, int idx, char *par)
     return;
   }
 
+#ifndef IPV6
+  if (!inet_pton(AF_INET, addr, saddr)) {
+    for (i = 0; addr[i]; i++) {
+      if (addr[i] == ':') {
+        dprintf(idx, "Invalid IP address format (this Eggdrop "
+          "was compiled without IPv6 support).\n");
+        return;
+      }
+    }
+  }
+#endif
+
+#ifndef TLS
+  if ((*port == '+') || (relay && (relay[1] == '+'))) {
+    dprintf(idx, "Ports prefixed with '+' are not enabled "
+      "(this Eggdrop was compiled without TLS support).\n");
+    return;
+  }
+#endif
+  if ((atoi(port) < 1) || (atoi(port) > 65535)) {
+    dprintf(idx, "Ports must be integers between 1 and 65535.\n");
+    return;
+  }
+  if (relay) {
+    relay2 = relay;
+    // Convert to just the port number string for error checking
+    relay2++;
+    if (*relay2 == '+') {
+      relay2++;
+    }
+    if ((atoi(relay2) < 1) || (atoi(relay2) > 65535)) {
+      dprintf(idx, "Ports must be integers between 1 and 65535.\n");
+      return;
+    }
+  }
+
+  if (*addr == '+') {
+    dprintf(idx, "Bot address may not start with a +.\n");
+    return;
+  }
+
+// Check if user forgot address field
+  for (i=0; i < strlen(addr); i++) {
+    if (!isdigit(addr[i]) && (addr[i] != '/')) {
```

Comment:
the isdigit function needs to be called with the character value (char) as an unsigned integer, like so:
isdigit((unsigned char)addr[i])
otherwise (on signed char platforms), negative values could be passed which is undefined behavior.
The reason isdigit() takes an int is that character constants ('a') are of type int.

Why is it making sure "addr" is only digits and / instead of the port field? addr would be the IP address, and port is already split into port and relayport and therefore checked.

Reply:
This checks to see if the address field was skipped by the user- if addr is all numbers (and /), it is not a valid address (neither IPv4, IPv6, or a hostname), and most likely the port field.